src/rust/engine/process_execution/src/nailgun/mod.rs

```diff
@@ -0,0 +1,254 @@
+use std::collections::btree_map::BTreeMap;
+use std::collections::btree_set::BTreeSet;
+use std::os::unix::fs::symlink;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::time::Duration;
+
+use futures::future::Future;
+use log::{debug, trace};
+
+use boxfuture::{Boxable, BoxFuture, try_future};
+use hashing::Digest;
+
+use crate::{ExecuteProcessRequest, ExecuteProcessRequestMetadata, FallibleExecuteProcessResult, MultiPlatformExecuteProcessRequest, Platform, Context};
+use crate::nailgun::nailgun_pool::NailgunProcessName;
+
+pub mod nailgun_pool;
+
+pub type NailgunPool = nailgun_pool::NailgunPool;
+
+// Hardcoded constants for connecting to nailgun
+static NAILGUN_MAIN_CLASS: &str = "com.martiansoftware.nailgun.NGServer";
+static ARGS_TO_START_NAILGUN: [&str; 1] = [":0"];
+
+// We can hardcode this because we mix it into the digest in the EPR.
```

Comment:
Add a TODO that this digest mixing-in should be done in the rust code, not the caller site